addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -88,8 +88,8 @@ export class DiscussCoreCommon {
             if (message.thread) {
                 const { selfMember } = message.thread;
                 if (
-                    message.id > selfMember?.seen_message_id.id &&
-                    notifId > selfMember.message_unread_counter_bus_id
+                    message.id > (selfMember?.seen_message_id?.id ?? 0) &&
+                    notifId > selfMember?.message_unread_counter_bus_id
                 ) {
```

Comment:
Ok for cautiousness with optional chaining. I think that if we have `selfMember.message_unread_counter_bus_id` then we necessarily have `selfMember.seen_message_id`, but I don't mind assuming less data knowledge in JS as this could theoretically happen.

Note that this works when no `selfMember`, but this is not necessarily intuitive that `positive integer / 0 > undefined` is always false (2nd condition prevents doing `selfMember.message_unread_counter` access when no `selfMember`).

Reply:
The case of having `selfMember.message_unread_counter_bus_id` and not having `selfMember.seen_message_id` happens in at least two cases:
- The conversation has no messages
- The message referenced by `selfMember.seen_message_id` was deleted

Reply:
Oh indeed, then still important to guard `seen_message_id` even with `message_unread_counter_bus_id`. Got it 👍